WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift

```diff
@@ -119,18 +119,23 @@ private extension InPersonPaymentsMenuViewController {
         permanentNoticePresenter.dismiss()
     }
 
-    func showOnboardingIfRequired() {
+    func showOnboarding() {
         guard let navigationController = self.navigationController else {
             return
         }
 
-        // Recreating it ensures that the readiness state is up to date.
-        // Keeping a reference ensures that the callback closure is retained.
-        cardPresentPaymentsOnboardingPresenter = CardPresentPaymentsOnboardingPresenter()
-
-        cardPresentPaymentsOnboardingPresenter?.showOnboardingIfRequired(from: navigationController) { [weak self] in
-            self?.cardPresentPaymentsOnboardingUseCase.refresh()
+        // Instead of using `CardPresentPaymentsOnboardingPresenter` we create the view directly because we already have the onboarding state in the use case.
+        // That way we avoid triggering the onboarding check again that comes with the presenter.
```

Comment:
Thanks for the comment here, this helped to understand the logic for these changes 👍